src/Utilities/OperatingSystemFilterHelpers.js

```diff
@@ -9,43 +10,76 @@ export const toGroupSelectionValue = (osVersions) => osVersions.reduce((acc, ver
-
-    return majorVersions.map((majorVersion) => ({
-        ...majorVersion,
-        groupSelectable: true,
-        items: osVersions.filter(({ value }) => {
-            const [majorOsVersion, minorOsVersion] = value.split('.');
-            return majorOsVersion === majorVersion.value.split('.')[0] && minorOsVersion !== '0';
-        }).sort(({ value }) => value.split('.')[1])
-    }));
+const compareVersions = (a, b, asc = true) =>
+    asc ? compare(coerce(a), coerce(b)) : rcompare(coerce(a), coerce(b));
+
+const groupOSVersions = (versions) => {
+    const groups = Object.entries(
+        versions.reduce((prev, { label, value }) => {
+            const major = value.split('.')[0];
+
+            if (prev[major] === undefined) {
+                prev[major] = {
+                    groupSelectable: true, // without this flag, the group won't be rendered - behavior of ConditionalFilter
+                    noFilter: true,
+                    label: 'RHEL ' + major,
+                    value: major + '.0',
+                    items: []
+                };
+            }
+
+            prev[major].items.push({
+                label,
+                value,
+                type: 'checkbox'
+            });
+
+            return prev;
+        }, {})
+    );
+
+    // sort by major versions in descending order
+    groups.sort((a, b) => compareVersions(a[0], b[0], false));
+
+    const sorted = groups.map((group) => {
+        // sort minor versions under each group
+        group[1].items.sort((a, b) => compareVersions(a.value, b.value, false));
+        return group[1];
+    });
+
+    return sorted;
 };
 
-export const buildOSFilterConfig = (config = {}) => ({
+export const buildOSFilterConfig = (config = {}, operatingSystems) => ({
     ...config,
     label: 'Operating System',
     value: 'os-filter',
     type: 'group',
     filterValues: {
         selected: config.value,
         onChange: config.onChange,
-        groups: groupOSVersions(operatingSystems).map((item) => ({
-            ...item,
-            type: 'checkbox',
-            items: item.items?.map((subItem) => ({
-                ...subItem,
-                type: 'checkbox'
-            }))
-        }))
+        groups: groupOSVersions(operatingSystems)
     }
 });
 
-export const buildOSChip = (operatingSystemValue) => {
-    const value = Object.values(operatingSystemValue || {}).flatMap((majorOsVersion) => Object.keys(majorOsVersion));
-    return (value?.length > 0 ? [{
-        category: 'Operating System',
-        type: OS_CHIP,
-        chips: operatingSystems.filter(({ value: osValue }) => value.some((osFilterValue) => osFilterValue === osValue))
-        .map(({ label, ...props }) => ({ name: label, ...props }))
-    }] : []);};
+export const buildOSChip = (operatingSystemValue = {}, operatingSystems) => {
+    const osGroups = Object.values(operatingSystemValue);
+    const minors = osGroups.flatMap((group) =>
```

Comment:
No need to assign to a value, you can directly use `Object.values(operatingSystemValue)`
```suggestion
    const minors = Object.values(operatingSystemValue).flatMap((group) =>
```